fairscale/nn/pipe/multiprocess_pipe.py

```diff
@@ -453,19 +312,16 @@ def __init__(
         self.pipelined_backward = pipelined_backward
         self.retain_graph = retain_graph
         self.pipeline: Optional[MultiProcessPipeline]
-        self.loss_fn = loss_fn
         self.lock = threading.Lock()
 
-        self.group = group
         self.worker_map = worker_map
         self.input_device = input_device
 
-        # The micro-batch index where the checkpointing stops.
-        checkpoint_stop = {"always": self.chunks, "except_last": self.chunks - 1, "never": 0}[self.checkpoint]
-
-        if self.group is None:
+        self.group: torch.distributed.ProcessGroup
+        if group is None:
             self.group = get_pipeline_parallel_group()
```

Comment:
can we make this the default value in line 282 above?

Reply:
get_pipeline_parallel_group() will assert is there isn't a model_parallel group.